### PR TITLE
[DOC]Detail explain flow.dag.yaml schema/concept: how to author nodes 

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: docs/how-to-guides/flow-yaml-schema.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
+# Flow YAML schema
```
[the remark this answers is not in the document]
Comment (YingChen1996) [reply docs/how-to-guides/flow-yaml-schema.md]: updated

File: docs/how-to-guides/develop-standard-flow.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
+# Develop standard flow
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-standard-flow.md:1]: I think we can have a develop-a-flow parent folder

File: docs/how-to-guides/develop-standard-flow.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+
+:::{admonition} Experimental feature
+This is an experimental feature, and may change at any time. Learn [more](faq.md#stable-vs-experimental).
+:::
+
+From this document, customer can learn how to develop the standard flow by writing a flow yaml from scratch. You can
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-standard-flow.md:7]: customer  -> you

Comment (wangchao1230) [inline docs/how-to-guides/develop-standard-flow.md:7]: develop "the" standard flow  -> develop a standard flow

File: docs/how-to-guides/develop-standard-flow.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+:sync: VS Code Extension
+When unfolding Inputs section in the authoring page, you can set and view your flow inputs, including input schema (name and type), 
+and the input value.
+
+![flow_input](../media/how-to-guides/develop-standard-flow/flow_input.png)
+:::
+
+::::
+For Web Classification sample as shown the screenshot above, the flow input is a url of string type. We also support 
+the input type of int, bool, double, list and object.
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-standard-flow.md:34]: add `` to the types. e.g. `int`

File: docs/how-to-guides/develop-standard-flow.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+The flow input data is the data that you want to process in your flow. 
+::::{tab-set}
+:::{tab-item} CLI
+:sync: CLI
+You can add a flow input in inputs section of flow yaml.
+```yaml
+inputs:
+  url:
+    type: string
+    default: https://www.microsoft.com/en-us/d/xbox-wireless-controller-stellar-shift-special-edition/94fbjc7h0h6h
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-standard-flow.md:20]: @crazygao @D-W- FYI

File: docs/how-to-guides/develop-standard-flow.md [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+
+![flow_input](../media/how-to-guides/develop-standard-flow/flow_input.png)
+:::
+
+::::
+For Web Classification sample as shown the screenshot above, the flow input is a url of string type. We also support 
+the input type of int, bool, double, list and object.
+
+## Develop the flow using different tools
+In one flow, you can consume different kinds of tools. We now support built-in tool like LLM, Python and Prompt and
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-standard-flow.md:37]: Give these tools reference a link

File: docs/how-to-guides/develop-standard-flow.md [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+
+::::
+
+### Edit tool
+::::{tab-set}
+:::{tab-item} CLI
+:sync: CLI
+You can edit the tool by simply open the source file and edit it. For example, we show simple Python tool code and LLM tool prompt below.
+
+```python
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-standard-flow.md:73]: put a sentence before the two samples

File: docs/how-to-guides/develop-standard-flow.md [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+    text_content: ${summarize_text_content.output} # Link LLM node with single-output upstream node
+```
+:::
+
+:::{tab-item} VS Code Extension
+:sync: VS Code Extension
+In the value drop-down, select `${inputs.url}`, `${prepare_examples.output}` and `${summarize_text_content.output}`, then 
+you'll see in the graph view that the newly created LLM node is linked to the flow input, upstream `prepare_examples` and `summarize_text_content` node. 
+
+![scenario1](../media/how-to-guides/develop-standard-flow/scenario1.png)
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-standard-flow.md:200]: Lets give the scenario a name, avoid using number in file name

File: docs/how-to-guides/develop-standard-flow.md [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+:sync: VS Code Extension
+
+![scenario3](../media/how-to-guides/develop-standard-flow/scenario3.png)
+:::
+
+::::
+When running the flow, the `input_str` input of the node will be replaced by flow input on the fly and the `input_str2` 
+input of the node will be replaced by `fetch_text_content_from_url` node output dictionary on the fly.
+
+## Set and check flow output
```
Comment (wangchao1230) [inline docs/how-to-guides/develop-standard-flow.md:273]: Set flow output whould be simpler I think.

File: docs/how-to-guides/develop-standard-flow.md (hunk 2 above)
Comment (wangchao1230) [inline docs/how-to-guides/develop-standard-flow.md:1]: Add links to this page in get-started.

File: docs/reference/flow-yaml-schema-reference.md [hunk 10]
```diff
@@ -0,0 +1,223 @@
+# Flow YAML Schema
+
+:::{admonition} Experimental feature
+This is an experimental feature, and may change at any time. Learn [more](../how-to-guides/faq.md#stable-vs-experimental).
+:::
+
+From this document, customer can get the details of flow schema.
```
Comment (wangchao1230) [inline docs/reference/flow-yaml-schema-reference.md:7]: we can remove this line

File: docs/reference/flow-yaml-schema-reference.md [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+The source JSON schema can be found at [Flow.schema.json](https://azuremlschemas.azureedge.net/promptflow/latest/Flow.schema.json)
+
+## YAML syntax
+
+| Key                        | Type      | Description                                                                                                                                                                                                       |
+|----------------------------|-----------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `$schema`                  | string    | The YAML schema. If you use the Azure Machine Learning VS Code extension to author the YAML file, including `$schema` at the top of your file enables you to invoke schema and resource completions.              |
```
Comment (wangchao1230) [inline docs/reference/flow-yaml-schema-reference.md:14]: Azure Machine Learning -> Prompt flow

File: docs/how-to-guides/develop-standard-flow.md (hunk 6 above)
Comment (YingChen1996) [reply docs/how-to-guides/develop-standard-flow.md]: added

File: docs/how-to-guides/develop-standard-flow.md (hunk 8 above)
Comment (YingChen1996) [reply docs/how-to-guides/develop-standard-flow.md]: updated

File: docs/how-to-guides/develop-standard-flow.md (hunk 2 above)
Comment (YingChen1996) [reply docs/how-to-guides/develop-standard-flow.md]: add to quik start and docs/index.md

File: docs/how-to-guides/develop-standard-flow.md (hunk 7 above)
Comment (YingChen1996) [reply docs/how-to-guides/develop-standard-flow.md]: updated

File: docs/how-to-guides/develop-standard-flow.md (hunk 2 above)
Comment (YingChen1996) [reply docs/how-to-guides/develop-standard-flow.md]: added

File: docs/reference/flow-yaml-schema-reference.md [hunk 12]
```diff
@@ -0,0 +1,223 @@
...
+
+
+### Flow input
+
+| Key             | Type                                      | Description                                | Allowed values                                      |
+|-----------------|-------------------------------------------|--------------------------------------------|-----------------------------------------------------|
+| `type`          | string                                    | The type of flow input.                    | `int`, `double`, `bool`, `string`, `list`, `object` |
+| `description`   | string                                    | Description of the input.                  |                                                     |
+| `default`       | int, double, bool, string, list or object | The default value for the input.           |                                                     |
+| `is_chat_input` | boolean                                   | Whether the input is the chat flow input.  |                                                     |
```
Comment (lalala123123) [inline docs/reference/flow-yaml-schema-reference.md:31]: A new property has been added, is_chat_history
type: boolean
description: Whether the input is the chat history for chat flow.

**Review by wangchao1230 — APPROVED** [review]

File: docs/reference/flow-yaml-schema-reference.md [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+
+## Examples
+
+Flow examples are available in the [GitHub repository](https://github.com/microsoft/promptflow/tree/main/examples/flows). 
+Below, we provide a web classification flow example.
+
+## YAML: Web classification flow YAML
+
+::::{tab-set}
+:::{tab-item} CLI
```
Comment (wangchao1230) [inline docs/reference/flow-yaml-schema-reference.md:78]: No need to have a tab here

File: docs/reference/flow-yaml-schema-reference.md [hunk 14]
```diff
@@ -0,0 +1,222 @@
...
+  source:
+    type: code
+    path: prepare_examples.py
+  inputs: {}
+- name: classify_with_llm
+  type: llm
+  source:
+    type: code
+    path: classify_with_llm.jinja2
+  inputs:
```
Comment (wangchao1230) [inline docs/reference/flow-yaml-schema-reference.md:137]: This is too long, maybe have a shorter one here

File: docs/reference/flow-yaml-schema-reference.md [hunk 15]
```diff
@@ -0,0 +1,222 @@
...
+| `default_variant_id` | string   | Default variant id.                                                                                                                                                                                                                                                                                           |                |
+| `variants `          | object   | This dictionary contains all node variations, with the variant id serving as the key and a node definition dictionary as the corresponding value.  Within the node definition dictionary, the key labeled 'node' should contain a variant definition similar to [Nodes](#nodes), excluding the 'name' field.  |                |
+
+
+
+## Examples
+
+Flow examples are available in the [GitHub repository](https://github.com/microsoft/promptflow/tree/main/examples/flows). 
+Below, we provide a web classification flow example.
```
Comment (wangchao1230) [inline docs/reference/flow-yaml-schema-reference.md:73]: maybe pick some classic example and put directly link here

File: docs/reference/flow-yaml-schema-reference.md (hunk 12 above)
Comment (wangchao1230) [reply docs/reference/flow-yaml-schema-reference.md:31]: @lalala123123  maybe directly contribute to this PR. 
As only have one page under develop-a-flow/ feels lonely. 
You can add your page develop-chat-flow together with this.

Or @YingChen1996 can create that empty page first

**Review by scurlate — APPROVED** [review]

File: docs/reference/flow-yaml-schema-reference.md (hunk 14 above)
Comment (YingChen1996) [reply docs/reference/flow-yaml-schema-reference.md:137]: this example contain may fields to help customer understand yaml schema. Maybe It's better than a shorter one?

Comment (YingChen1996) [reply docs/reference/flow-yaml-schema-reference.md:137]: I will remove the yaml here and just put some sample links here.

File: docs/reference/flow-yaml-schema-reference.md (hunk 12 above)
Comment (YingChen1996) [reply docs/reference/flow-yaml-schema-reference.md:31]: added

File: docs/reference/flow-yaml-schema-reference.md (hunk 15 above)
Comment (YingChen1996) [reply docs/reference/flow-yaml-schema-reference.md:73]: added

**Review by wangchao1230 — APPROVED** [review]

**Review by brynn-code — APPROVED** [review]

**Pull request merged**